samples/boards/stm32/power_mgmt/standby_shutdown/src/main.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/pm/pm.h>
+#include <stm32_ll_pwr.h>
+
+#if !defined(CONFIG_SOC_SERIES_STM32L4X)
+#error Not implemented for other series
+#endif  /* CONFIG_SOC_SERIES_STM32L4X */
+
+#define STACKSIZE 1024
+#define PRIORITY 7
+#define SLEEP_TIME_MS   3000
+
+#define SW0_NODE	DT_ALIAS(sw0)
+#if !DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#error "Unsupported board: sw0 devicetree alias is not defined"
+#endif
+
+/* Semaphore used to control button pressed value */
+static struct k_sem button_sem;
+
+static int led_is_on;
+
+static const struct gpio_dt_spec button =
+	GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, {0});
+
+static const struct gpio_dt_spec led =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+static struct gpio_callback button_cb_data;
+
+void config_wakeup_features(void)
+{
+	/* Configure wake-up features */
+	/* WKUP2(PC13) only , - active low, pull-up */
+	/* Set pull-ups for standby modes */
+	LL_PWR_EnableGPIOPullUp(LL_PWR_GPIO_C, LL_PWR_GPIO_BIT_13);
+	LL_PWR_IsWakeUpPinPolarityLow(LL_PWR_WAKEUP_PIN2);
+	/* Enable pin pull up configurations and wakeup pins */
+	LL_PWR_EnablePUPDCfg();
+	LL_PWR_EnableWakeUpPin(LL_PWR_WAKEUP_PIN2);
+	/* Clear wakeup flags */
+	LL_PWR_ClearFlag_WU();
+}
+
+void button_pressed(const struct device *dev, struct gpio_callback *cb,
+		    uint32_t pins)
+{
+	k_sem_give(&button_sem);
+}
+
+void thread_Shutdown_Standby_mode(void)
+{
+	k_sem_init(&button_sem, 0, 1);
+	k_sem_take(&button_sem, K_FOREVER);
+	gpio_pin_configure(led.port, led.pin, GPIO_DISCONNECTED);
+	printk("User button pressed\n");
+	config_wakeup_features();
+	if (led_is_on == false) {
+		printk("Shutdown Mode requested\n");
+		printk("Release the user button to exit from Shutdown Mode\n\n");
+#ifdef CONFIG_LOG
+		k_msleep(2000);
+#endif /* CONFIG_LOG */
```

Comment:
2 sec seems quite long for user. Likely user would release the button before that.
Also 2 sec seems large to print a 30 char line, what about 200 ms  ?